### PR TITLE
Generate: get generation mode from the generation config instance 🧼

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

 This PR moves `get_generation_mode` from a stand-alone function requiring a `generation_config` to a `generation_config` method. Also adds some missing documentation to public functions 🤗 

🧼 This PR adds no new code, except for a new test.

### CODE REVIEW

File: docs/source/en/main_classes/text_generation.md [hunk 1]
```diff
@@ -37,6 +37,9 @@ like token streaming.
 	- from_pretrained
 	- from_model_config
 	- save_pretrained
+	- update
+	- validate
+	- get_generation_mode
```
Comment (gante) [inline docs/source/en/main_classes/text_generation.md:42]: Documents important functions that I expect users to use in their projects

**Review by ArthurZucker — APPROVED** [review]
Thanks for adding this, very convenient ! 
It could even be a property WDYT? 
This way printing the generation config always tells you which mode you will be using

**Comment by gante** [comment]
> This way printing the generation config always tells you which mode you will be using

That's a cool idea!

**Comment by gante** [comment]
@ArthurZucker Actually I'm going to leave it as a todo -- because of assisted generation, whose trigger is not an attribute of the config, we can't store it cleanly as a property. Going to focus on more relevant feature for now :D

**Pull request merged**